### PR TITLE
Add more info about weather and refactor

### PR DESCRIPTION
**Pull request by AndrewHelix**

### CODE REVIEW

File: src/types/weatherTypes.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  sunsetEpoch: number;
+  moonrise: string;
+  moonset: string;
+  moonPhase: number;
+  uvIndex: number;
+  minVisibility: number;
+  pressure: number;
+  confidence: string;
+  solarRadiationSum: number;
+}[];
```
Comment (JuliaShavyrova) [inline src/types/weatherTypes.ts]: looks like these interfaces have duplicative properties. please move them into separate interface

File: src/components/main/Main.tsx [hunk 2]
```diff
@@ -4,37 +4,33 @@ import commonStyle from '../../styles/commonStyles.css';
...
+  const [dailyWeather, setDailyWeather] = useState<DailyWeatherType>([]);
 
   async function fetchData(lon: number, lat: number) {
-    await getCity(lon, lat).then(locationData => setCity(locationData.city));
-    await getCityWeather(lon, lat).then(cityWeather => setWeather(cityWeather));
+    Promise.all([
+      getCity(lon, lat),
+      getWeather<CurrentWeatherType>('/api/v1/current/', lon, lat),
+      getWeather<{ forecast: HourlyWeatherType }>('/api/v1/forecast/hourly/', lon, lat),
+      getWeather<{ forecast: DailyWeatherType }>('/api/v1/forecast/daily/', lon, lat)
```
Comment (JuliaShavyrova) [inline src/components/main/Main.tsx]: why do you store in state HourlyWetherType and DailyWeatherType as plain objects, but CurrentWeatherType is stored as object with next structure? do you need `current` property?
```
 {
  current: {
    ...
  }
}
```

File: src/components/main/MainCard.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    date: date.getDate(),
+    month: date.getMonth() + 1,
+    year: date.getFullYear(),
+    hours: date.getHours(),
+    minutes: date.getMinutes()
+  };
+};
+
+const MainCard: FC<MainCardProps> = ({ currentWeather, location }) => {
+  const date = convertTime(currentWeather.current.time);
```
Comment (JuliaShavyrova) [inline src/components/main/MainCard.tsx]: the same question, why not just pass `currentWeather.current` as prop?

File: src/components/main/defaultStates.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
+export const defaultCurrentWeather = {
```
Comment (JuliaShavyrova) [inline src/components/main/defaultStates.ts]: need type definition here

File: src/components/main/Main.tsx (hunk 2 above)
Comment (AndrewHelix) [reply src/components/main/Main.tsx]: No, i don't need `current`
My bad. I'll fix it )

File: src/components/main/defaultStates.ts (hunk 4 above)
Comment (AndrewHelix) [reply src/components/main/defaultStates.ts]: ok:)
but if I have default state, which don't mathes with type, which defined in generic (like this `useState<CurrentWeatherType>(defaultCurrentWeather)`) I will have a mistake and can't compilate project.

File: src/components/main/MainCard.tsx [hunk 5]
```diff
@@ -21,7 +21,7 @@ const convertTime = (timeData: string) => {
 };
 
 const MainCard: FC<MainCardProps> = ({ currentWeather, location }) => {
```
Comment (JuliaShavyrova) [inline src/components/main/MainCard.tsx:23]: not required to be fixed in current PR, but would be better to use destructuring for `currentWeather` prop in this case

**Review by JuliaShavyrova — APPROVED** [review]

File: src/components/main/defaultStates.ts (hunk 4 above)
Comment (JuliaShavyrova) [reply src/components/main/defaultStates.ts]: that's why we are using typescript - to catch such errors :)

**Pull request merged**